src/pages/Signup/SignupPage.jsx

```diff
@@ -0,0 +1,341 @@
+import React, { useState, useEffect } from "react";
+import { Formik } from "formik";
+import * as Yup from "yup";
+import { Link, Link as RouterLink } from "react-router-dom";
+import { useMutation, gql } from "@apollo/client";
+
+import {
+  Box,
+  Button,
+  FormControl,
+  IconButton,
+  Grid,
+  InputLabel,
+  OutlinedInput,
+  Stack,
+  Typography,
+  InputAdornment,
+  FormHelperText,
+} from "@mui/material";
+
+import VisibilityIcon from "@mui/icons-material/Visibility";
+import VisibilityOffIcon from "@mui/icons-material/VisibilityOff";
+
+const SIGNUP_MUTATION = gql`
+  mutation Signup(
+    $firstName: String!
+    $lastName: String!
+    $dob: String!
+    $email: String!
+    $password: String!
+    $address: String!
+  ) {
+    signup(
+      firstName: $firstName
+      lastName: $lastName
+      dob: $dob
+      email: $email
+      password: $password
+      address: $address
+    ) {
+      id
+      firstName
+      lastName
+      dob
+      email
+      password
+      address
+    }
+  }
+`;
```

Comment:
Thsi will probably need to change but the core functionality is there and code looks good - Just need to update the fields for what we want new members to look like - We can sync on this